internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -218,10 +246,15 @@ func (r *delegatingResolver) updateClientConnStateLocked() error {
 	}
 	var addresses []resolver.Address
 	for _, targetAddr := range (*r.targetResolverState).Addresses {
-		addresses = append(addresses, proxyattributes.Set(proxyAddr, proxyattributes.Options{
-			User:        r.proxyURL.User,
-			ConnectAddr: targetAddr.Addr,
-		}))
+		// Avoid proxy when network is not tcp.
+		if networkType, ok := networktype.Get(targetAddr); ok && networkType != "tcp" {
+			addresses = append(addresses, targetAddr)
+		} else {
+			addresses = append(addresses, proxyattributes.Set(proxyAddr, proxyattributes.Options{
+				User:        r.proxyURL.User,
+				ConnectAddr: targetAddr.Addr,
+			}))
+		}
```

Comment:
nit: You can use `continue` in the `if` block to avoid indenting the `else` block.